### PR TITLE
Feature/task 5080 5088 new info screen for notes

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
This PR is the [TASK 5088 ](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5088)for [feature 5080 ](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5080)contact diary enhancements. It adds the note information screen.

image

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	private lazy var navigationFooterItem: ENANavigationFooterItem = {
+		let item = ENANavigationFooterItem()
+
+		item.primaryButtonTitle = AppStrings.NewVersionFeatures.buttonContinue
+		item.isPrimaryButtonEnabled = true
+		item.isSecondaryButtonHidden = true
+
+		// [KGA] remove
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:47]: 👀

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 2]
```diff
@@ -1080,6 +1080,11 @@ enum AppStrings {
 				}
 			}
 		}
+		
+		enum NotesInformation {
+			static let title = NSLocalizedString("ContactJournalNotesDescriptionTitle", comment: "")
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:1085]: Underscores are missing.

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift (hunk 1 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift]: Ooooh damn.... 😄

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: Adjusted 👍

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+//
+
+import UIKit
+
+class DiaryDayNotesInfoViewController: DynamicTableViewController, DismissHandling {
+
+	// MARK: - Attributes
+
+	var finished: (() -> Void)?
+
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:12]: We don't have a MARK for attributes.
If it's required for the DeltaOnboardingViewControllerProtocol please move it to a dedicated section

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	private let viewModel: DiaryDayNotesInfoViewModel
+
+	private func setupRightBarButtonItem() {
+		navigationItem.rightBarButtonItem = CloseBarButtonItem(
+			onTap: { [weak self] in
+				self?.finished?()
+			}
+		)
+	}
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:72]: we have a new logic for the closeButton - please have look inside the DismissHandling protocol.
just add ```navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton``` inside the ```init()``` method after ```super.init()``` and you are done

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+			forCellReuseIdentifier: CustomCellReuseIdentifiers.roundedCell.rawValue
+		)
+
+		dynamicTableViewModel = viewModel.dynamicTableViewModel
+	}
+}
+
+// MARK: - Cell reuse identifiers.
+
+extension DiaryDayNotesInfoViewController {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:89]: private ?

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:12]: Thanks. I corrected it. It was not used at all.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift (hunk 5 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift]: Should be 👍  changed it.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift (hunk 4 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift]: Adjusted according to DidmissHandling 👍

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		item.isPrimaryButtonEnabled = true
+		item.isSecondaryButtonHidden = true
+
+		return item
+	}()
+	
+	// MARK: - Protocol DismissHandling
+	
+	func wasAttemptedToBeDismissed() {
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:49]: Todo?

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift (hunk 6 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift:49]: you can remove the whole method, just declare conformance to the protocol is enough

**Pull request merged**